src/phoenix/trace/tracer.py

```diff
@@ -100,13 +100,22 @@ def create_span(
             conversation=conversation,
         )
 
+        self.add_span(span)
+
+        return span
+
+    def add_span(self, span: Span) -> None:
+        """Adds a span to the tracer.
+
+        Args:
+            span (Span): The span to be add to the tracer.
+        """
         if self._exporter:
             self._exporter.export(span)
         self.span_buffer.append(span)
```

Comment:
may want to hold a lock here if we want to be extra safe (`self._lock = RLock()`) since the stream may be called by a separate thread
```suggestion
        with self._lock:
            self.span_buffer.append(span)
```